syntax = "proto3";
//只能用这种命名方式 不能用小写 下划线
package GameProto;

//服务器类型
enum ServerType
{
	ST_NONE = 0;
	ST_Login = 1;
	ST_Team = 2;
	ST_GameManager = 3;
}

//player->login
message PlayerRequestLogin
{
	uint64 qw_player_id = 1;
	string sz_nick_name = 2;
	string sz_avatar = 3;
	uint32 dw_sex = 4;
	uint32 dw_balance = 5;
	string sz_token = 6;
}

//login->player
message LoginAckPlayerLoginResult
{
	uint32 dw_result = 1;
}

//game->gamemanager
message GameNotifyGameManagerInfo
{
	uint32 dw_player_port = 1;
	uint32 dw_server_port = 2;
	uint32 dw_slave_server_port = 3;
	uint64 qw_player_point = 4;
}

//gamemanager->game
message GameManagerAckGameInfoResult
{
	uint32 dw_result = 1;
}

//player->gamemanager
message PlayerRequestGameManagerInfo
{
	uint32 dw_result = 1;
}

//gamemanager->player
message GameManagerAckPlayerInfoResult
{
	uint32 dw_player_port = 1;
	uint32 dw_server_port = 2;
	uint32 dw_slave_server_port = 3;
}

//player->game
message PlayerRequestGameTest
{
	string sz_test = 1;
}

//serverinfo(server->server_center, server_center->server)
message ServerInfo
{
	uint32 dw_server_id = 1;
	string sz_listen_ip = 2;
	uint32 dw_team_port = 3;
	uint32 dw_game_server_manager_port = 4;
	uint32 dw_login_port = 5;
}
